dask/sizeof.py

```diff
@@ -1,10 +1,15 @@
+import logging
 import random
 import sys
 from array import array
 from distutils.version import LooseVersion
 
+import pkg_resources
```

Comment:
Can we avoid the top-level import of `pkg_resources`? It's a large module and tends to slow down imports. Or is the idea that importing dask will give you the full set of registered sizeof implementations?

This also implies a runtime dependency on setuptools. We would need to add that to our requirements if we're using pkg_resources. There's also the [entrypoints](https://pypi.org/project/entrypoints/) package, which I've heard is nicer to work with.

Reply:
> Or is the idea that importing dask will give you the full set of registered sizeof implementations?

I think that's the idea here

FWIW there also appears to be a new `importlib.metadata` module in the standard library (added in Python 3.8) which is recommended by the PyPA (https://packaging.python.org/guides/creating-and-discovering-plugins/#using-package-metadata). Note there's also a `importlib_metadata` backport package for older versions of Python. 

cc'ing @jakirkham in case you have any recommendations here

Reply:
Thanks for the comments all.

I agree that we should avoid `pkg_resources`. I didn't do so initially as this was a starting point for discussion. I think `importlib.metadata` might be the better option here, but I haven't benchmarked vs `entrypoints`.


> This seems like a nice enhancement and sorry I hadn't gotten a chance to look at it yet. It'd be helpful if you could write some documentation and maybe open a WIP PR on a downstream library that implements sizeof entrypoints. That would help with discussion.

In the description I link two example projects that make use of this hook. Are they sufficient for explanation at this point? Let me know if I can do more here!